contracts/service-registry/src/client.rs

```diff
@@ -0,0 +1,169 @@
+use error_stack::ResultExt;
+use router_api::ChainName;
+
+use crate::msg::{ExecuteMsg, QueryMsg};
+use crate::{Service, Verifier, WeightedVerifier};
+
+type Result<T> = error_stack::Result<T, Error>;
+
+#[derive(thiserror::Error, Debug, PartialEq)]
+pub enum Error {
+    #[error("failed query the service registry for active verifiers for service {service_name} and chain {chain_name}")]
+    QueryActiveVerifiers {
+        service_name: String,
+        chain_name: ChainName,
+    },
+
+    #[error("failed to query service registry for service {0}")]
+    QueryService(String),
+
+    #[error("failed to query service registry for verifier {verifier} of service {service_name}")]
+    QueryVerifier {
+        service_name: String,
+        verifier: String,
+    },
+}
+
+impl<'a> From<client::Client<'a, ExecuteMsg, QueryMsg>> for Client<'a> {
+    fn from(client: client::Client<'a, ExecuteMsg, QueryMsg>) -> Self {
+        Client { client }
+    }
+}
+
+pub struct Client<'a> {
+    client: client::Client<'a, ExecuteMsg, QueryMsg>,
+}
+
+impl<'a> Client<'a> {
```

Comment:
suggestion for the error handling:
implement `From` for the QueryMsg for conversion, then you can do something like:
```
let msg = ...;
self.client.query(&msg).change_context_lazy(|| msg.into())
```